GTMAppAuthSwift/Sources/AuthStateStore.swift

```diff
@@ -0,0 +1,99 @@
+  /// - Note: This save operation will use the `itemName` provided during initialization.
+  @objc func save(authState: AuthState) throws
+
+  /// Removes the `AuthState`.
+  ///
+  /// - Throws: Any error that may arise during the removal.
+  /// - Note: This removal will use the `itemName` provided during initialization.
+  @objc func removeAuthState() throws
+
+  /// Retrieves the `authState`.
+  ///
+  /// - Throws: Any error that may arise during the retrieval.
+  /// - Note: This retrieval will use the `itemName` provided during initialization.
+  @objc func retrieveAuthState() throws -> AuthState
+}
+
+/// Represents the operations needed to convert OAuth2 authorizations to `AuthState` instances along
+/// with credential storage.
+@objc public protocol GTMOAuth2AuthStateStore {
+  /// Attempts to create an `AuthState` from stored data in GTMOAuth2 format, at the supplied
+  /// identifier.
+  ///
+  /// - Parameters:
+  ///   - itemName: The name for the item to find.
+  ///   - tokenURL: The OAuth token endpoint URL.
+  ///   - redirectURI: The OAuth redirect URI used when obtaining the original authorization.
+  ///   - clientID: The OAuth client ID.
+  ///   - clientSecret: The OAuth client secret.
+  /// - Returns: An `AuthState` object.
+  /// - Throws: Any error arising from the `AuthState` creation.
+  @objc func retrieveAuthStateInGTMOAuth2Format(
+    forItemName itemName: String,
+    tokenURL: URL,
+    redirectURI: String,
+    clientID: String,
+    clientSecret: String?
+  ) throws -> AuthState
+
+  /// Attempts to create a `AuthState` from data stored in a GTMOAuth2 format at the the supplied
+  /// identifier.
+  ///
+  /// Uses Google OAuth provider information.
+  ///
+  /// - Parameters:
+  ///   - itemName: The keychain name.
+  ///   - clientID: The OAuth client id.
+  ///   - clientSecret: The OAuth client secret.
+  /// - Returns: An `AuthState` object, or nil.
+  /// - Throws: Any error arising from the `AuthState` creation.
+  @objc func retrieveAuthStateForGoogleInGTMOAuth2Format(
+    forItemName itemName: String,
+    clientID: String,
+    clientSecret: String
+  ) throws -> AuthState
+
+  /// Saves the authorization state in a GTMOAuth2 compatible manner.
+  ///
+  /// - Parameters:
+  ///   - authorization: The `AuthState` to save.
+  ///   - itemName: The name of the item to save.
+  /// - Throws: An instance of `GTMOAuth2KeychainCompatibility.Error` arising from the save.
+  @available(*, deprecated, message: "Use AuthStateStore.save(authState:)")
+  @objc func saveWithGTMOAuth2Format(
+    forAuthorization authorization: AuthState,
+    withItemName itemName: String
+  ) throws
+
+  /// Removes stored tokens, such as when the user signs out.
+  ///
+  /// - Parameters:
+  ///   - itemName: The name of the item to remove.
+  /// - Throws: An instance of `GTMOAuth2KeychainCompatibility.Error` arising from the removal.
+  @objc func removeOAuth2AuthState(withItemName itemName: String) throws
```

Comment:
Are you saying that this overlaps with `AuthStateStore.removeAuthState`? If so, I would say yes because `GTMOAuth2AuthStateStore` is a different protocol.